qiskit_optimization/applications/ising/base_application.py

```diff
@@ -0,0 +1,23 @@
+import copy
+from abc import ABC, abstractmethod
+
+
+class BaseApplication(ABC):
```

Comment:
Let's rename to `BaseOptimizationApplication` so we can also have different ones.